var/spack/repos/builtin/packages/silo/package.py

```diff
@@ -57,6 +57,13 @@ def flag_handler(self, name, flags):
 
     @when('%clang@9:')
     def patch(self):
+        self.clang_9_patch()
+
+    @when('%apple-clang@11.0.3:')
+    def patch(self):
+        self.clang_9_patch()
+
+    def clang_9_patch(self):
```

Comment:
This is hackier than I would like. I think it would be more intuitive to add the patch for all versions, then check the compiler and exit if the patch isn't needed.

Reply:
Changed. But I will point out that leads to spack messages that it's running the patch phase for specs that don't require a patch.

Reply:
Let's defer this to a later PR, but I agree with @becker33 that it might confuse people to see that a patch function is run (and does nothing).

Reply:
I never ended up pushing my changes since I borked my branch. I think @alalazo and I are on the same page here.